Common/CostFunctions/itkAdvancedImageToImageMetric.h

```diff
@@ -168,9 +166,8 @@ class ITK_TEMPLATE_EXPORT AdvancedImageToImageMetric : public ImageToImageMetric
   using BSplineOrder2TransformPointer = typename BSplineOrder2TransformType::Pointer;
   using BSplineOrder3TransformPointer = typename BSplineOrder3TransformType::Pointer;
 
-  /** Typedefs for multi-threading. */
-  using ThreaderType = itk::PlatformMultiThreader;
```

Comment:
As you see here, originally it said `ThreaderType = itk::PlatformMultiThreader`, so without thread pools! 😞

Reply:
so, does this last one mean that we were *not* using threadpools for the metric, before this PR?

Reply:
> so, does this last one mean that we were _not_ using threadpools for the metrics, before this PR?

No, not exactly. We were _already_ using threadpools for various metrics. At least for AdvancedMeanSquaresImageToImageMetric and SumSquaredTissueVolumeDifferenceImageToImageMetric. Because these two metrics call `AdvancedImageToImageMetric::LaunchGetValueThreaderCallback()`, which uses the thread-pool support of ITK's `ImageToImageMetric`.

On the other hand, PCAMetric_F_multithreaded did explicitly instantiate `auto local_threader = ThreaderType::New()`, so it was using ThreaderType (= PlatformMultiThreader), which does not use threadpools.